### PR TITLE
Correct duplicate primary key error

### LINKED ISSUE #44: 500 error with query returned from trying to add duplicate DID allowance
In the `POST allow/publish-did/{DID}` endpoint if a duplicate DID allowance is attempted a 500 error in the following form returns

```
{
  "detail": "(sqlalchemy.dialects.postgresql.asyncpg.IntegrityError) <class 'asyncpg.exceptions.UniqueViolationError'>: duplicate key value violates unique constraint \"allowedpublicdid_pkey\"\nDETAIL:  Key (registered_did)=(123) already exists.\n[SQL: INSERT INTO allowedpublicdid (registered_did) VALUES (%s) RETURNING allowedpublicdid.created_at, allowedpublicdid.updated_at]\n[parameters: ('123',)]\n(Background on this error at: https://sqlalche.me/e/14/gkpj)"
}
```

Would handle this with a cleaner 4xx level error I think.

### PR DESCRIPTION
**Pull request by Gavinok**
Resolves #44 by catching db integrity errors and changing them into a HTTP 409 (conflict) error rather than a HTTP 500

### CODE REVIEW

**Comment by Gavinok** [comment]
Some small dependency changes where necessary in order to get both tests and the endorser running.

**Comment by loneil** [comment]
Were there changes needed to get this working/building (regardless of specific changes for the feature here)?

I see a number of large dependency bumps recently https://github.com/hyperledger/aries-endorser-service/pulls?q=is%3Apr+is%3Aclosed
so wondering if `main` was in a state where it was working prior to this?

File: endorser/requirements.txt [hunk 1]
```diff
@@ -29,9 +29,9 @@ sniffio==1.2.0
 SQLAlchemy==1.4.27
 sqlalchemy2-stubs==0.0.2a19
 sqlmodel==0.0.6
-starlette==0.36.2
+starlette==0.35.1
```
Comment (loneil) [inline endorser/requirements.txt:32]: Downgrading this necessary to get it building?

File: endorser/api/services/allow_lists.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                await endorse_transaction(db, transaction)
+    except Exception as e:
+        # try to retrieve and print text on error
+        logger.error(f"Failed to update pending transactions {e}")
+
+
+B = TypeVar("B", bound=BaseModel)
+
+
+async def add_to_allow_list(db: AsyncSession, a: B) -> B:
```
Comment (esune) [inline endorser/api/services/allow_lists.py:48]: Any reason to not use `BaseModel` directly rater than type-bind `B` to it?
Wondering about readability in the longer term, when we get back to the code and forgot what this is supposed to do 😅

**Review by esune — COMMENTED** [review]
Changes overall look good, just a small comment on my end.

@loneil raises good points, however if we are sure code will work after doing a pass through dependencies we may be able to ignore past dependabot-driven updates?

File: endorser/requirements.txt (hunk 1 above)
Comment (Gavinok) [reply endorser/requirements.txt:32]: Unfortunately yes. I could try to upgrading the dependency requiring the downgrade.

File: endorser/api/services/allow_lists.py (hunk 2 above)
Comment (Gavinok) [reply endorser/api/services/allow_lists.py:48]: Just to make mypy/pyright happy. This ensures that the class of `a` is identical to the return type of this function and not another class deriving from BaseModel. For example if we changed the function to
```python
async def add_to_allow_list(db: AsyncSession, a: BaseModel) -> BaseModel:
```
You get this error when calling `add_to_allow_list` in allow.py
```
Pyright [reportReturnType]: Expression of type "BaseModel" is incompatible with return type "AllowedCredentialDefinition"
```
While not as intuitive it's the proper way to enforce this

**Comment by Gavinok** [comment]
@loneil I have ran the E2E regression tests and these changes seem to be working without issue. Not sure if that is what you had in mind

**Comment by loneil** [comment]
> @loneil I have ran the E2E regression tests and these changes seem to be working without issue. Not sure if that is what you had in mind

If the BDD end-to-end is good enough coverage to rely on then yeah.

File: endorser/requirements.txt (hunk 1 above)
Comment (loneil) [reply endorser/requirements.txt:32]: Ah yeah looks like FastAPI version restricts the starlette version

**Comment by esune** [comment]
> > @loneil I have ran the E2E regression tests and these changes seem to be working without issue. Not sure if that is what you had in mind
> 
> If the BDD end-to-end is good enough coverage to rely on then yeah. I don't know enough about the coverage of those tests but it might be enough?

I think they should be good enough for now, as the project evolves we may add-on to them, but currently the functionality is relatively simple.

**Review by esune — APPROVED** [review]

**Pull request merged**